File: python/BUILDER_COMPOSITE/BUILDER_COMPOSITE/builder/ConstructorReservacionVIP.py
from ..Reservacion.ReservacionVIP import ReservacionVIP
from .ConstructorReservacion import ConstructorReservacion

class ConstructorReservacionVIP(ConstructorReservacion):
    
    def __init__(self) -> None:
        self.reservacionVIP = ReservacionVIP()
    
    def construirDatosCliente(self, nombres,
                              apellidoPaterno,
                              apellidoMaterno,
                              edad,
                              tipoCliente):
        self.reservacionVIP.datosCliente.setDatosCliente(nombres,
                              apellidoPaterno,
                              apellidoMaterno,
                              edad,
                              tipoCliente)
    
    def construirDatosCompra(self, numeroTarjeta,
                             fechaVencimiento):
        self.reservacionVIP.datosCompra.setDatosCompra(numeroTarjeta,
                             fechaVencimiento)
    
    
    def construirDatosFuncion(self, asiento,
                              horario,
                              pelicula):
        self.reservacionVIP.datosFuncion.setDatosFuncion(asiento,
                              horario,
                              pelicula)
    
    def construirDatosBeneficios(self, beneficios):
        self.reservacionVIP.datosBeneficios.setDatosBeneficios(beneficios)
    
    def obtenerReservacion(self):
        return self.reservacionVIP

File: php/BUILDER_COMPOSITE/constructores/ConstructorReservacionVIP.php
<?php

require_once 'ConstructorReservacion.php';
require_once './reservaciones/ReservacionVIP.php';
class ConstructorReservacionVIP implements ConstructorReservacion{
    public $reservacionVIP;
    
    public function __construct() {
        $this->reservacionVIP = new ReservacionVIP();
    }
    
    public function construirDatosCliente($nombre, $apellidoP, $apellidoM, $edad, $tipoCliente) {
        $this->reservacionVIP->getDatosCliente()->setDatosCliente($nombre, $apellidoP, $apellidoM, $edad, $tipoCliente);
    }

    public function construirDatosCompra($numTarjeta, $fecha) {
        $this->reservacionVIP->getDatosCompra()->setDatosCompra($numTarjeta, $fecha);
    }

    public function construirDatosFuncion($asiento, $hora, $dia, $mes, $anio, $pelicula) {
        $this->reservacionVIP->getDatosFuncion()->setDatosFuncion($asiento, $hora, $dia, $mes, $anio, $pelicula);
    }
    
    public function construirDatosBeneficio($beneficios){
        $this->reservacionVIP->getDatosBeneficio()->setDatosBeneficios($beneficios);
    }

    public function obtenerReservacion() {
        return $this->reservacionVIP;
    }

}
